src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,222 @@
+                        return@stop
+                    }
+                }
+            }
+        }
+
+        return Observable.just(MangasPage(mangas, false))
+    }
+
+    // Not efficient, but the simplest way for me to refresh
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        fetchPopularManga(0)
+            .map { mangasPage ->
+                mangasPage
+                    .mangas
+                    .first { it.url == manga.url }!!
+            }
+
+    // This implementation here is needlessly complicated, for it has to automatically detect
+    // whether we’re in a date-annotated archive, the main archive, or a dateless archive.
+    // All three are largely similar, there are just *some* (annoying) differences we have to
+    // deal with.
+    private fun fetchChaptersForDatedAndDatelessArchives(archivePages: Document, archivePath: String): List<SChapter> {
+        val chapters = mutableListOf<SChapter>()
+        var i = 0
+        archivePages.select("""div.text > table.text > tbody > tr""").forEach {
+            val pageData = it.select("""td""")
+            var pageAnchor = pageData.getOrNull(2)?.selectFirst("a")
+            // null for »Intermission«, main archive, dateless archive,…
+            if (pageAnchor != null) {
+                val a = pageAnchor
+
+                chapters.add(
+                    SChapter.create().apply {
+                        name = a.text()
+                        chapter_number = i.toFloat()
+                        date_upload = DATE_FMT.parse(pageData[0].text().trim())?.time ?: 0L
+                        url = absolutePageUrl(a.attr("href"), archivePath)
+                    },
+                )
+                i += 1
+            } else if (!pageData.hasAttr("colspan")) {
+                // Are we in a dateless archive?
+                pageAnchor = pageData.getOrNull(0)?.selectFirst("a")
+                if (pageAnchor != null) {
+                    // For now we assume all dateless archives were published the same day
+                    // the Solo book was published, because currently Solo’s the only one such book.
+                    val pageDate = 1660435200000L + (i * 1000 * 60)
+                    val a = pageAnchor
+
+                    chapters.add(
+                        SChapter.create().apply {
+                            name = a.text()
+                            chapter_number = i.toFloat()
+                            date_upload = pageDate
+                            url = absolutePageUrl(a.attr("href"), archivePath)
+                        },
+                    )
+                    i += 1
+                }
+            }
+        }
+        chapters.reverse()
+        return chapters
+    }
+
+    private fun absolutePageUrl(pageUrl: String, maybeRelativeTo: String): String =
+        if (pageUrl.startsWith('/')) {
```

Comment:
This should be resolved for you if you use `link.absUrl("href")` or `link.attr("abs:href")`, it will resolve the relative path based on the document's `.location()` (set when using `.asJsoup()` (ParsedHttpSource does this for you))

Reply:
Oh, good to know. Will use that, much cleaner.